### ISSUE TITLE
Is there a user manual ?

### ISSUE THREAD

**Issue by gbdomubpkm** [Julow/Unexpected-Keyboard#463]
Hello.

Thank you for this keyboard that I discovered and tested since yesterday but I have difficulty understanding how it works and having access to its full potential.

Is there a user manual or link that I can't find ?

For example, it's basic, but how do you do accented characters?

Please tell me what I need to know to use this keyboard properly.

Best regards.

**Comment by Julow** [comment]
There's no documentation unfortunately. I would love to have some in the app and more in the README but I don't have the time to write it.

You might see symbols like this:
![2023-09-03-14-43](https://github.com/Julow/Unexpected-Keyboard/assets/2310568/cdcf5351-57c4-466e-a1b0-e9ddacc3e49d)

These are dead-keys: You press them first, then the accented letter.

If you don't see these, the keyboard might not know your language. You can go to the "Add keys to the keyboard" setting to add them.
What's your language ?

**Comment by sdrapha** [comment]
I though about doing some video clips, that would show how it work easier than puting it into words. Also no translation required.
I will try to make some when I have a chance, @Julow then you can decide later if they are useful or not, and where to put them, there are so many options. Google play, github readme, a link to it in the app settings. (I would not embbed the clips on the app though, to keep the apk tiny in size)

**Comment by Julow** [comment]
That would be awesome :)

**Comment by gbdomubpkm** [comment]
Hello @Julow Hello.

I finally figured out (not easily)  how to do the accents before your comment.

In use, for the moment, everything that is punctuation (including the dot!), apostrophe, comma is not easy for me to find on the keyboard. I would like you to be able to choose to put bold/color characters in the touch angles you want. If you wish, I will make  a feature request.

Best regards

**Comment by Julow** [comment]
There are some issues about the symbols in the corners: https://github.com/Julow/Unexpected-Keyboard/issues/400 https://github.com/Julow/Unexpected-Keyboard/issues/226

**Comment by Ytrog** [comment]
Is there a list with the compose key combinations somewhere?

I would love a cheat-sheet for them. If there is no cheat-sheet, but a list sonewhere then I might make one 🤔

**Comment by Julow** [comment]
Yes `en_US_UTF_8_Compose.pre` is the list but not every lines are used. `<Multi_key>` is the compose-key itself, only line starting with that are taken into account for the compose key. Also, some combinations are not taken due to limitations in the implementation.

This list is parsed in a quick and dirty way using this [python script](https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/compose/compile.py), which generates this [state machine](https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/ComposeKeyData.java). You might be able to modify the python script to list all the combinations instead. It's run with `gradle compileComposeSequences`.
There's currently 2013 sequences but many are duplicates (eg. `'e` and `e'`) and many can be grouped (eg. `'` followed by almost any letter).

**Comment by Ytrog** [comment]
I wonder if a directed graph representation would be useful?

For example for ŋ you could have

```dot
digraph compose {

Compose -> n;
n -> ŋ;
```

Which you clould then render with Graphviz 🤔

Of course this is just an example with a single entry, but one with all possibilities would be nice I think.

**Comment by Ytrog** [comment]
@Julow Made this sample (only for lowercase n combinations) on my phone using Graphviz in Termux.

I such a format useful to persue?

![compose.png](https://github.com/Julow/Unexpected-Keyboard/assets/2880376/1dd75cf7-5df3-426a-a1e6-3bbc3e81fad2)

**Comment by Julow** [comment]
The transition character are missing but I'm afraid it might look impossibly complicated with 2000 arrows. Perhaps with the "Compose" node removed and a separate graph for each starting character ? Perhaps subgraphs could help visually separate all the graphs ?
